fhir2dataset/parser.py

```diff
@@ -0,0 +1,193 @@
+import re
+from collections import defaultdict, OrderedDict
+
+
+PREFIX = [" eq ", " ne ", " gt ", " lt ", " ge ", " le ", " sa ", " eb ", " ap "]
+
+
+class FHIR2DatasetParser:
+    def __init__(self):
+        self.CLAUSES = OrderedDict()
+        self.__split_mask_clauses = ""
+
+        self.__init_clauses()
+        self.__init_split_mask()
+        self.__reset_config()
+
+    def parse(self, SQL_string: str):
+        self.__reset_config()
+        SQL_string = SQL_string.replace("\n", " ")
+        # print(f"SQL_string: {SQL_string}")
+
+        items = re.split(self.__split_mask_clauses, SQL_string)
+        items.remove("")
+        for clause in self.CLAUSES.keys():
+            for idx, item in enumerate(items):
+                if item == clause:
+                    # print(f"item: {item}")
+                    self.CLAUSES[item](items[idx + 1])
+
+        self.__create_config()
+        return self.config
+
+    def __create_config(self):
+        self.config["from"] = dict(self.__from)
+        self.config["select"] = dict(self.__select)
+        self.config["where"] = dict(self.__where)
+        if self.__inner_join:
+            if not self.config["join"]:
+                self.config["join"] = dict()
```

Comment:
You can put that one out of the if statement to factorise, because you clean empty entries at the end anyway

Reply:
Duly noted! I removed the if at this point and modified the `__create_config` function so that the joined key is removed if its dictionary is `{"inner":{},"child":{},"parent":{}`.